mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -466,17 +473,19 @@ def create_model_version(
             )
         )
         with tempfile.TemporaryDirectory() as tmpdir:
-            try:
-                local_model_dir = mlflow.artifacts.download_artifacts(
-                    artifact_uri=source, dst_path=tmpdir, tracking_uri=self.tracking_uri
-                )
-            except Exception as e:
-                raise MlflowException(
-                    f"Unable to download model artifacts from source artifact location "
-                    f"'{source}' in order to upload them to Unity Catalog. Please ensure "
-                    f"the source artifact location exists and that you can download from "
-                    f"it via mlflow.artifacts.download_artifacts()"
-                ) from e
+            local_model_dir = local_model_path
+            if local_model_dir is None:
+                try:
+                    local_model_dir = mlflow.artifacts.download_artifacts(
+                        artifact_uri=source, dst_path=tmpdir, tracking_uri=self.tracking_uri
+                    )
+                except Exception as e:
+                    raise MlflowException(
+                        f"Unable to download model artifacts from source artifact location "
+                        f"'{source}' in order to upload them to Unity Catalog. Please ensure "
+                        f"the source artifact location exists and that you can download from "
+                        f"it via mlflow.artifacts.download_artifacts()"
+                    ) from e
```

Comment:
Can we remove `local_model_dir` and only use `local_model_path`? Using both seems confusing.